### PR TITLE
[16.0][FIX] intrastat refunds.

### PR DESCRIPTION
**Pull request by ljsalvatierra-factorlibre**
Modify transaction ID for the following cases:
* Dispatches:
  * out_invoice B2B: 11
  * out_invoice B2C: 12
  * in_refund: 21
* Arrivals:
  * out_refund: 21
  * in_invoice: 11 Don't propagate intrastat transaction to refund.
Add some tests.
Remove unused files.

Related PR in l10n-spain: https://github.com/OCA/l10n-spain/pull/3322

### CODE REVIEW

**Review by aritzolea — APPROVED** [review]
Code review

**Review by MayteGLC — APPROVED** [review]
LGTM

**Comment by ljsalvatierra-factorlibre** [comment]
Hi, could you please review this @pedrobaeza @victoralmau ? And its related PR https://github.com/OCA/l10n-spain/pull/3322

File: intrastat_product/wizards/account_move_reversal.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+
+class AccountMoveReversal(models.TransientModel):
+    _inherit = "account.move.reversal"
+
+    def _prepare_default_reversal(self, move):
+        res = super()._prepare_default_reversal(move)
+        res.update(
+            {
+                "intrastat_transport_id": False,
```
Comment (pedrobaeza) [inline intrastat_product/wizards/account_move_reversal.py:13]: Explain why these changes, and also add docstring putting this reason/explanation for documenting the code.

File: intrastat_product/models/intrastat_product_declaration.py [hunk 2]
```diff
@@ -308,16 +308,26 @@ def _get_partner_country(self, inv_line, notedict, eu_countries):
...
         transaction = invoice.intrastat_transaction_id
-        fp2transaction = {
-            "b2b": self.env.ref("intrastat_product.intrastat_transaction_11"),
-            "b2c": self.env.ref("intrastat_product.intrastat_transaction_12"),
-        }
-        if not transaction:
-            # as we have searched with intrastat_fiscal_position in ('b2b', 'b2c')
-            # we should always have intrastat_fiscal_position in fp2transaction
-            transaction = fp2transaction[invoice.intrastat_fiscal_position]
+        intr_fp = invoice.intrastat_fiscal_position
```
Comment (pedrobaeza) [inline intrastat_product/models/intrastat_product_declaration.py:317]: Why using fiscal position?

File: intrastat_product/wizards/account_move_reversal.py (hunk 1 above)
Comment (SoniaViciana) [reply intrastat_product/wizards/account_move_reversal.py:13]: The change is based on -> https://sede.agenciatributaria.gob.es/Sede/aduanas/intrastat-obligaciones-estadisticas/noticias-interes/nuevos-codigos-naturaleza-transaccion-intrastat.html

On the other hand, I don't understand why code 12 doesn't match with this other documentation... -> https://www.intrastat.com/aeat/normativa/intrastat/tabla-naturaleza-de-la-transaccion-para-intrastat.php

File: intrastat_product/models/intrastat_product_declaration.py (hunk 2 above)
Comment (SoniaViciana) [reply intrastat_product/models/intrastat_product_declaration.py:317]: Because in fiscal position you can define if it is code 11 or 12:

imagen

**Comment by ljsalvatierra-factorlibre** [comment]
Hi @pedrobaeza could you review please the comments from @SoniaViciana ?

File: intrastat_product/models/intrastat_product_declaration.py (hunk 2 above)
Comment (pedrobaeza) [reply intrastat_product/models/intrastat_product_declaration.py:317]: OK

File: intrastat_product/wizards/account_move_reversal.py (hunk 1 above)
Comment (pedrobaeza) [reply intrastat_product/wizards/account_move_reversal.py:13]: Please document this in code.
